engineering/conventions-go.md

```diff
@@ -0,0 +1,50 @@
+
+# Go Development Conventions
+
+## Introduction
+
+This guide documents development conventions for Go at source{d}. Check [general conventions](conventions.md) for language-independent conventions that are also applicable for Go projects.
+
+## Supported Go Versions
+
+* Support latest two stable major versions of Go (e.g. 1.9.x and 1.10.x).
+  * Some of our projects may support more versions, specially when there is wide-adoption outside source{d} (e.g. [go-git](https://github.com/src-d/go-git) currently supports three).
+
+## Dependency Management
+
+* If your project is an application:
+  * Use [dep](https://github.com/golang/dep) to manage dependencies.
+  * Check in the `vendor` directory into git.
+* If your project is a library:
+  * You may use [dep](https://github.com/golang/dep) to manage dependencyes if needed. If you do, do not commit `vendor` directory.
+
+## Build System
+
+* If your project requires fetching or building dependencies, implement that in the `dependencies` rule.
+* If binaries can be built, do so in the `packages` rule. Note that [src-d/ci](https://github.com/src-d/ci) provides that rule by default.
+
+## Testing
+
+* Use [testify](https://github.com/stretchr/testify).
+* Use [suites](https://github.com/stretchr/testify#suite-package) to share setup/teardown code among related tests. When testing multiple methods of the same struct, you generally want to start with a suite.
+
+## Executables
+
+* Put sources for your executable commands in `cmd/<your command>`.
+* Use [go-flags](https://github.com/jessevdk/go-flags) for CLI options parsing.
```

Comment:
Why we use it instead of Cobra, by the way? I already hit few bugs in it while trying to work on bblfshd. It seems far less stable.

Reply:
I think we just use it because of repetition from one project to the next. I also found such problems and had better experience with other more popular CLI lbraries...

It would be nice to use [cobra](https://github.com/spf13/cobra) next time. In any case, since I don't mean to introduce changes with this initial PR, I've just rephrased this point to be less prescriptive.

A PR can be made later if we change to cobra, either by using it on a specific project, or proposing it for wider adoption.

Reply:
We use go-flags because I like more, I used cobra in some personal project, and I don't like too much. 

I like more `go-flags` is way less more simple, and easy to work having the flags as fields of the structs. So I vote to use `go-flags`, for two reasons, is already in many of our projects, and second because I think it's better.

Reply:
If we continue to go this way, we should definitely send few PRs to fix it at least.